ldap/servers/plugins/replication/cl5_api.c

```diff
@@ -2365,23 +2352,23 @@ _cl5PurgeGetNextEntry(CL5Entry *entry, void *iterator, DBT *key)
 
         return rc;
     }
-    slapi_ch_free(&(data.data));
+    dblayer_value_free(it->it_cldb->be, &data);
```

Comment:
Not a complaint, just a question about why the value free needs the be pointer now? Just want to know more about how this works :)

Reply:
data is a dbi_val_t (i.e a struct) and as usual for struct, you need to refer it through its address ...
 FYI: I do not simply free data.data because data.flags determine whether the buffer need to be free or not.

Reply:
Right I see. So in this case &data is informing if the dblayer_value_free should free the be pointer?

Reply:
If by "the pointer" you means data,data then yes 

int dblayer_value_free(Slapi_Backend *be __attribute__((unused)), dbi_val_t *data)
{   
    if (data != NULL && !(data->flags & DBI_VF_PROTECTED)) {
        slapi_ch_free(&data->data);
        data->size = data->ulen = 0;
    }
    return DBI_RC_SUCCESS;
}